src/artemis/devices/eiger.py

```diff
@@ -174,7 +174,9 @@ def arm_detector(self):
         bit_depth = self.bit_depth.get()
         self.odin.file_writer.data_type.put(bit_depth)
 
-        set_and_wait(self.odin.file_writer.capture, 1, timeout=10)
+        odin_status = self.odin.file_writer.capture.set(1)
+        odin_status &= await_value(self.odin.meta.ready, 1)
```

Comment:
You should wait for OD:Capture_RBV here as well, which I guess `set` does not do.

Reply:
both the `set` method and the `await_value` methods here return status objects (futures), and the `wait` method below then waits for both futures to complete

Reply:
Ah. Put with callback on Capture returns when the file is closed, not when it has finished opening, so that may not do what you want.

Reply:
my bad, looks like I was looking at the wrong method in ophyd when I wrote that edit. The `set` method calls `set_and_wait` in a separate thread, and then returns a future which completes once the read PV is correct. So using `set` here should work.